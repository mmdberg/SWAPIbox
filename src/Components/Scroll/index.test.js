import React from 'react';
import { shallow } from 'enzyme';
import { Scroll } from './index';

describe('Scroll', () => {

    it('should match the snapshot', () => {
      let mockOpening = {
        text: "It is a period of civil war. Rebel spaceships.",
        title: "A New Hope",
        releaseYear: "1977"
       };
      let wrapper = shallow(<Scroll opening={mockOpening}/>)

      expect(wrapper).toMatchSnapshot()
    });
});