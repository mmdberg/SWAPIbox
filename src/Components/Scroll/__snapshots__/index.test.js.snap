// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scroll should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    opening={
        Object {
            "releaseYear": "1977",
            "text": "It is a period of civil war. Rebel spaceships.",
            "title": "A New Hope",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="scroll"
>
          <div
                    className="scroll-text"
          >
                    <p>
                              It is a period of civil war. Rebel spaceships.
                    </p>
                    <h3>
                              A New Hope
                    </h3>
                    <h3>
                              1977
                    </h3>
          </div>
          <div
                    className="fade"
          />
</div>,
        <Link
          className="enter-site"
          replace={false}
          to="/home"
>
          Enter for Star Wars Data
</Link>,
      ],
      "className": "welcome",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="scroll-text"
>
              <p>
                            It is a period of civil war. Rebel spaceships.
              </p>
              <h3>
                            A New Hope
              </h3>
              <h3>
                            1977
              </h3>
</div>,
            <div
              className="fade"
/>,
          ],
          "className": "scroll",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  It is a period of civil war. Rebel spaceships.
</p>,
                <h3>
                  A New Hope
</h3>,
                <h3>
                  1977
</h3>,
              ],
              "className": "scroll-text",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "It is a period of civil war. Rebel spaceships.",
                },
                "ref": null,
                "rendered": "It is a period of civil war. Rebel spaceships.",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "A New Hope",
                },
                "ref": null,
                "rendered": "A New Hope",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1977",
                },
                "ref": null,
                "rendered": "1977",
                "type": "h3",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fade",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Enter for Star Wars Data",
          "className": "enter-site",
          "replace": false,
          "to": "/home",
        },
        "ref": null,
        "rendered": "Enter for Star Wars Data",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="scroll"
>
            <div
                        className="scroll-text"
            >
                        <p>
                                    It is a period of civil war. Rebel spaceships.
                        </p>
                        <h3>
                                    A New Hope
                        </h3>
                        <h3>
                                    1977
                        </h3>
            </div>
            <div
                        className="fade"
            />
</div>,
          <Link
            className="enter-site"
            replace={false}
            to="/home"
>
            Enter for Star Wars Data
</Link>,
        ],
        "className": "welcome",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="scroll-text"
>
                <p>
                                It is a period of civil war. Rebel spaceships.
                </p>
                <h3>
                                A New Hope
                </h3>
                <h3>
                                1977
                </h3>
</div>,
              <div
                className="fade"
/>,
            ],
            "className": "scroll",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    It is a period of civil war. Rebel spaceships.
</p>,
                  <h3>
                    A New Hope
</h3>,
                  <h3>
                    1977
</h3>,
                ],
                "className": "scroll-text",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "It is a period of civil war. Rebel spaceships.",
                  },
                  "ref": null,
                  "rendered": "It is a period of civil war. Rebel spaceships.",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "A New Hope",
                  },
                  "ref": null,
                  "rendered": "A New Hope",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "1977",
                  },
                  "ref": null,
                  "rendered": "1977",
                  "type": "h3",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fade",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Enter for Star Wars Data",
            "className": "enter-site",
            "replace": false,
            "to": "/home",
          },
          "ref": null,
          "rendered": "Enter for Star Wars Data",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
